apps/web/src/components/Onboarding/SetupPassword/SetupPassword.tsx

```diff
@@ -101,14 +120,16 @@ export const SetupPassword = ({ mode }: SetupPasswordProps) => {
           });
           break;
         }
+        default:
+          break;
       }
 
       return onClose();
     });
 
   const icon = isNewMnemonic ? UserIcon : LockIcon;
-  const title = isNewMnemonic ? "Create Password" : "Almost there";
-  const buttonLabel = isNewMnemonic ? "Create Account" : "Import Wallet";
+  const title = mode ? (isNewMnemonic ? "Create Password" : "Almost there") : "Confirm password";
```

Comment:
if we have so much logic going on, we might consider splitting the component into several separate components to simplify the logic & testing. right now, you have quite a few branches and they all depend on the mode only